test/e2e/helm_chart_test.go

```diff
@@ -0,0 +1,134 @@
+						AddRole("readWrite", "Ships", "").
+						WithAuthDatabase("admin"),
+				},
+			)
+			userSpec.Clusters = append(userSpec.Clusters, model.LoadUserClusterConfig("data/atlascluster_basic_helm.yaml"))
+			userSpec.Clusters[0].Spec.Project.Name = userSpec.Project.GetK8sMetaName()
+			userSpec.Clusters[0].ObjectMeta.Name = "cluster-from-helm" // helm template has equal ObjectMeta.Name and Spec.Name
+			userSpec.Clusters[0].Spec.Name = "cluster-from-helm"
+		})
+		By("User use helm for deploying operator", func() {
+			helm.AddMongoDBRepo()
+			helm.InstallKubernetesOperatorNS(userSpec)
+		})
+		deployCluster(userSpec, 30006)
+	})
+
+	It("[helm-wide] User can deploy operator namespaces by using HELM", func() {
+		By("User creates configuration for a new Project and Cluster", func() {
+			userSpec = model.NewUserInputs(
+				"only-key",
+				[]model.DBUser{
+					*model.NewDBUser("reader2").
+						WithSecretRef("dbuser-secret-u2").
+						AddRole("readWrite", "Ships", "").
+						WithAuthDatabase("admin"),
+				},
+			)
+			userSpec.Clusters = append(userSpec.Clusters, model.LoadUserClusterConfig("data/atlascluster_basic_helm.yaml"))
+			userSpec.Clusters[0].Spec.Project.Name = userSpec.Project.GetK8sMetaName()
+			userSpec.Clusters[0].ObjectMeta.Name = "cluster-from-helm-wide" // helm template has equal ObjectMeta.Name and Spec.Name
+			userSpec.Clusters[0].Spec.Name = "cluster-from-helm-wide"
+		})
+		By("User use helm for deploying operator", func() {
+			helm.AddMongoDBRepo()
+			helm.InstallKubernetesOperatorNS(userSpec)
+		})
+		deployCluster(userSpec, 30007)
+	})
+
+})
+
+func deployCluster(userSpec model.UserInputs, appPort int) {
+	By("User deploy cluster by helm", func() {
+		kube.CreateApiKeySecret(userSpec.KeyName, userSpec.Namespace)
+		helm.InstallCluster(userSpec)
+	})
+	By("Wait creation until is done", func() {
+		waitProject(userSpec, "1")
+		userSpec.ProjectID = kube.GetProjectResource(userSpec.Namespace, userSpec.K8sFullProjectName).Status.ID
+		waitCluster(userSpec, "1")
+	})
+
+	By("Check attributes", func() {
+		uCluster := mongocli.GetClustersInfo(userSpec.ProjectID, userSpec.Clusters[0].Spec.Name)
+		compareClustersSpec(userSpec.Clusters[0].Spec, uCluster)
+	})
+
+	By("check database users Attibutes", func() {
+		Eventually(checkIfUsersExist(userSpec), "2m", "10s").Should(BeTrue())
+		checkUsersAttributes(userSpec)
+	})
+
+	By("Deploy application for user", func() {
+		// kube apply application
+		// send data
+		// retrieve data
+		checkUsersCanUseApplication(appPort, userSpec)
+	})
+
+	By("Check project, cluster does not exist", func() {
+		helm.Uninstall(userSpec.Clusters[0].Spec.Name, userSpec.Namespace)
+		Eventually(
+			func() bool {
+				return mongocli.IsProjectInfoExist(userSpec.ProjectID)
+			},
+			"5m", "20s",
+		).Should(BeFalse(), "Project and cluster should be deleted from Atlas")
+	})
+
+	By("Delete HELM release", func() {
+		helm.UninstallKubernetesOperatorNS(userSpec)
```

Comment:
Can we check that the resources in Atlas were removed eventually?

Reply:
The operator itself? in Atlas, we check cluster/project deletion